src/Riok.Mapperly/Emit/SourceEmitter.cs

```diff
@@ -11,10 +11,14 @@ public static class SourceEmitter
 {
     private const string AutoGeneratedComment = "// <auto-generated />";
 
-    public static CompilationUnitSyntax Build(MapperDescriptor descriptor, CancellationToken cancellationToken)
+    public static CompilationUnitSyntax Build(
+        MapperDescriptor descriptor,
+        bool instanceWithStaticPartialMethod,
+        CancellationToken cancellationToken
+    )
     {
         var ctx = new SourceEmitterContext(
-            descriptor.Symbol.IsStatic,
+            descriptor.Symbol.IsStatic || instanceWithStaticPartialMethod,
```

Comment:
Instead of using this additional boolean parameter, why not just store it inside the descriptor as a new property (eg. `public bool Static { get; set; }`)?

Reply:
Make sense, I do not know why I chose `instanceWithStaticPartialMethod`